src/pycsou/abc/operator.py

```diff
@@ -1632,6 +1633,8 @@ def __init__(self, shape: pyct.NonAgnosticShape):
         """
         super(LinOp, self).__init__(shape)
         self._diff_lipschitz = 0
+        if shape[0] == shape[1]:
+            self.__class__ = SquareOp
```

Comment:
Does this really work ? I'm having second thoughts about this automatic recasting of ``LinOps`` with identical domain/co-domain dimensions... It's probably not very recommended. The cleanest would probably be to keep the trace method in ``SquareOp`` and let the user instantiate from the right class if he wants to benefit from the ``trace`` method. @joanrue, @SepandKashani some thoughts?